### ISSUE TITLE
Consider adding `IsUpgradeError` func for readability 

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#3821]
## Summary

see [comment](https://github.com/cosmos/ibc-go/pull/3791#discussion_r1226479830)

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by crodriguezvega** [comment]
Do we still need this and do we actually need to check that the error is an `UpgradeError` in the message server? The error is casted before a call to `MustAbortUpgrade` [here](https://github.com/cosmos/ibc-go/blob/3203f07f10c76d17e094773754f5da10c49933c5/modules/core/keeper/msg_server.go#L771) and [here](https://github.com/cosmos/ibc-go/blob/3203f07f10c76d17e094773754f5da10c49933c5/modules/core/keeper/msg_server.go#L824), but `abortUpgrade` already does the casting [here](https://github.com/cosmos/ibc-go/blob/3203f07f10c76d17e094773754f5da10c49933c5/modules/core/04-channel/keeper/upgrade.go#L840) and if the error is not an `UpgradeError`, then one is constructed anyway. So I think that in the message server we can just do (e.g. in `ChannelUpgradeAck`):

```go
err = k.ChannelKeeper.ChanUpgradeAck(ctx, msg.PortId, msg.ChannelId, msg.CounterpartyFlushStatus, msg.CounterpartyUpgrade, msg.ProofChannel, msg.ProofUpgrade, msg.ProofHeight)
if err != nil {
  ctx.Logger().Error("channel upgrade ack failed", "error", errorsmod.Wrap(err, "channel upgrade ack failed"))
  k.ChannelKeeper.MustAbortUpgrade(ctx, msg.PortId, msg.ChannelId, upgradeErr)

  // NOTE: a FAILURE result is returned to the client and an error receipt is written to state.
  // This signals to the relayer to begin the cancel upgrade handshake subprotocol.
  return &channeltypes.MsgChannelUpgradeAckResponse{Result: channeltypes.FAILURE}, nil
}
```

which is basically the same that it is done if an error is returned from the callback.

**Comment by crodriguezvega** [comment]
Closed by #4144.
